packages/block-library/src/block/index.php

```diff
@@ -41,13 +41,31 @@ function render_block_core_block( $attributes ) {
 
 	$seen_refs[ $attributes['ref'] ] = true;
 
+	$filter_block_context = static function( $context ) use ( $attributes ) {
+		if ( isset( $attributes['dynamicContent'] ) && $attributes['dynamicContent'] ) {
+			$context['dynamicContent'] = $attributes['dynamicContent'];
```

Comment:
For the block context, it would be nice to name it in a way that created a connection with the pattern, e.g., `pattern/dynamicContent`.